content/blog/2023/08/02/2023-08-23-gitlab-plugin-modernization-report.adoc

```diff
@@ -0,0 +1,89 @@
+---
+:layout: post
+:title: "GSoC GitLab Plugin Modernization Project"
+:tags:
+- gsoc
+- gsoc2023
+- gitlab
+- plugins
+:author: harsh-ps-2003
+:sig: gsoc
+description: >
+  The GitLab Plugin Modernization is a project under Google Summer of Code (GSoC) 2023.
+opengraph:
+  image: /images/gsoc/opengraph.png
+links:
+  gitter: https://app.gitter.im/#/room/#jenkinsci_gitlab-plugin:gitter.im
+---
+
+
+The goal of this blog is to showcase the work done on GitLab Plugin during the Google Summer of Code 2023.
+For a detailed description of the project, please see the link:/projects/gsoc/2023/projects/gitlab-plugin-modernization/[project page].
+
+---
+
+== Overview
+
+. <<About Project>>
+. Coding <<Phase 1>>
+. Coding <<Phase 2>>
+. <<Future Improvements>>
+. <<Acknowledgments>>
+. <<Useful Links>>
+
+---
+
+== About Project
+
+Plugins grows old and becomes difficult to maintain with time. GitLab Plugin was one such plugin that needed some love and care. The GitLab Plugin Modernization project thus aimed to modernize the GitLab Plugin by the use of old (link:https://resteasy.dev[RESTEasy]) library with modern (link:https://github.com/gitlab4j/gitlab4j-api[GitLab4J-API]) library via Jenkins (link:https://plugins.jenkins.io/gitlab-api/[GitLab API plugin]), normalizing the GitLab plugin’s use of the GitLab API with regard to the (link:https://plugins.jenkins.io/gitlab-oauth/[GitLab Authentication]), (link:https://plugins.jenkins.io/gitlab-branch-source/[GitLab Branch Source]), and (link:https://plugins.jenkins.io/gitlab-logo/[GitLab Logo]) plugins. This modernization reduces future maintenance burden and improves the overall quality of the plugin by making it more lightweight due to reduced dependencies and enhanced documentation.
+
+The migration was performed in the following 2 coding phases:
+
+---
+
+== Phase 1
+
+In this phase, the following tasks were completed:
+
+* Migration of GitLab Plugin from RESTEasy to GitLab4J-API via the GitLab-API plugin, without affecting the webhook functionality of the plugin. (link:[PR])
+* Interactive testing of partially migrated plugin.
+
+The achievenments of this phase were demonstrated in the (link:https://youtu.be/W4eSVCTmqb8[Midterm Presentation])
+
+---
+
+== Phase 2
+
+In this phase, the following tasks were completed:
+* Adapting webhooks implementation to GitLab4J-API.(link:https://github.com/jenkinsci/gitlab-plugin/pull/1553[PR])
+* Migrating the Proxy Settings from RESTEasy to GitLab4J-API.(link:https://github.com/jenkinsci/gitlab-plugin/pull/1566[PR])
+* Partially adapting the Unit and Integration tests.
+* Improving the documentation of the plugin.(link:https://github.com/jenkinsci/gitlab-plugin/pull/1556[PR])
+* Interactive testing of the fully migrated plugin.
+
+The results of this phase will be showcased in in the Final Presentation in the future.
```

Comment:
```suggestion
In this phase, the following tasks were completed:

* link:https://github.com/jenkinsci/gitlab-plugin/pull/1553[Adapting webhooks implementation to GitLab4J-API].
* link:https://github.com/jenkinsci/gitlab-plugin/pull/1566[Migrating the proxy settings from RESTEasy to GitLab4J-API].
* Partially adapting the Unit and Integration tests.
* link:https://github.com/jenkinsci/gitlab-plugin/pull/1556[Improving the documentation of the plugin].
* Interactive testing of the fully migrated plugin.

The results of this phase will be showcased in in the final presentation in the future.
```
suggestion for formatting/making links part of the text by adding them to the in line text as opposed to after, removing () from links as they are not necessary.